src/app/auth/services/user.service.ts

```diff
@@ -62,8 +63,9 @@ const deleteUserAvatar = (): Promise<void> => {
 };
 
 const sendVerificationEmail = (): Promise<void> => {
-  const usersClient = SdkFactory.getInstance().createUsersClient();
-  return usersClient.sendVerificationEmail();
+  const usersClient = SdkFactory.getNewApiInstance().createUsersClient();
+  const token = localStorageService.get('xNewToken') || undefined;
+  return usersClient.sendVerificationEmail(token);
```

Comment:
if you can add some tests at least for this function to cover all the code little by little